### PR TITLE
Replace sleeps with `wait_for_*`

### LINKED ISSUE #593: Replace sleep calls with wait_for_*
Calling sleep in tests is flaky because it assumes that 2 or 3 or 10 seconds is enough for something to happen.
However, it could cause flakiness in tests depending on the environment running the tests.

I suggest replacing sleep calls where applicable to:
- `wait_for_l2_block`
- `wait_for_l1_block`
- `wait_for_l1_commitment_submitted`
- `wait_for_soft_confirmation`
- ... etc

### Acceptance criteria
We don't use `sleep` to wait for any node in test to reach a specific block / state.

### PR DESCRIPTION
**Pull request by rakanalh**
# Description
Replace sleep with `wait_for_*`
## Linked Issues
- Fixes #593 
- Depends on #626

### CODE REVIEW

File: bin/citrea/tests/all_tests.rs [hunk 1]
```diff
@@ -10,3 +10,4 @@ mod test_helpers;
 
 const DEFAULT_MIN_SOFT_CONFIRMATIONS_PER_COMMITMENT: u64 = 1000;
 const DEFAULT_DEPOSIT_MEMPOOL_FETCH_LIMIT: usize = 10;
+const DEFAULT_PROOF_WAIT_DURATION: u64 = 300; // 4 minutes
```
Comment (eyusufatik) [inline bin/citrea/tests/all_tests.rs:13]: 5 mins :)

File: bin/citrea/tests/test_helpers/mod.rs [hunk 2]
```diff
@@ -156,8 +162,96 @@ pub fn tempdir_with_children(children: &[&str]) -> TempDir {
...
-        std::fs::create_dir(p).unwrap();
+        if !std::path::Path::new(&p).exists() {
+            std::fs::create_dir(p).unwrap();
+        }
     }
 
     db_dir
 }
+
+pub async fn wait_for_l2_batch(sequencer_client: &TestClient, num: u64, timeout: Option<Duration>) {
```
Comment (eyusufatik) [inline bin/citrea/tests/test_helpers/mod.rs:173]: let's call wait_for_l2_block

just like the l1 one

File: bin/citrea/tests/test_helpers/mod.rs [hunk 3]
```diff
@@ -156,8 +162,96 @@ pub fn tempdir_with_children(children: &[&str]) -> TempDir {
...
+        let now = SystemTime::now();
+        if start + timeout <= now {
+            panic!("Timeout");
+        }
+
+        sleep(Duration::from_secs(1)).await;
+    }
+}
+
+pub async fn wait_for_commitment(
```
Comment (eyusufatik) [inline bin/citrea/tests/test_helpers/mod.rs:236]: i think we should indicate this checks commitments on postgres, the name should reflect that

**Review by eyusufatik — CHANGES_REQUESTED** [review]
great enhancement to the codebase 👍

**Review by eyusufatik — APPROVED** [review]

File: bin/citrea/tests/sequencer_commitments/mod.rs [hunk 4]
```diff
@@ -10,18 +10,18 @@ use sov_modules_api::{BlobReaderTrait, SignedSoftConfirmationBatch};
...
+    create_default_sequencer_config, start_rollup, tempdir_with_children, wait_for_l1_block,
+    wait_for_l2_block, wait_for_postgres_commitment, wait_for_prover_l1_height, NodeMode,
 };
-use crate::DEFAULT_DEPOSIT_MEMPOOL_FETCH_LIMIT;
+use crate::{DEFAULT_DEPOSIT_MEMPOOL_FETCH_LIMIT, DEFAULT_PROOF_WAIT_DURATION};
 
 #[tokio::test]
 async fn sequencer_sends_commitments_to_da_layer() {
-    // citrea::initialize_logging();
+    citrea::initialize_logging();
```
Comment (kpp) [inline bin/citrea/tests/sequencer_commitments/mod.rs:24]: This should be commented out

File: bin/citrea/tests/sequencer_commitments/mod.rs [hunk 5]
```diff
@@ -371,7 +377,7 @@ async fn test_ledger_get_commitments_on_slot() {
 
 #[tokio::test]
 async fn test_ledger_get_commitments_on_slot_prover() {
-    // citrea::initialize_logging();
+    citrea::initialize_logging();
```
Comment (kpp) [inline bin/citrea/tests/sequencer_commitments/mod.rs:380]: as well

**Pull request merged**